src/hammers/image_deployer.py

```diff
@@ -269,19 +269,35 @@ def get_image_build_timestamp(image):
             raise Exception(f"Invalid build_timestamp format: {build_timestamp}")
 
 
-def archive_image(image_connection, image):
+def archive_image(image_connection,
+                  image,
+                  image_metadata_field="chameleon-supported"):
     logging.debug(f"Renaming existing image {image.name}.")
     archive_date = get_image_build_timestamp(image)
+    archived_name = f"{image.name}_{archive_date}"
     image_connection.image.update_image(
         image.id,
-        name=f"{image.name}_{archive_date}"
+        name=archived_name,
     )
-
-    archived_name = f"{image.name}_{archive_date}"
     logging.debug(f"Renamed image {image.name} to {archived_name}.")
 
+    action = f"{image_metadata_field}=no"
+    logging.debug(f"Setting '{action}' on image {image.id}.")
+    try:
+        image_connection.compute.set_image_metadata(
+            image.id,
+            **{image_metadata_field: "no"}
```

Comment:
for my reading comprehension:
this is the place where we're setting supported=no on archived images?
and the rest is just a refactor of making the metadata field name configurable?

Reply:
yep, that's correct